pkg/component/component_test.go

```diff
@@ -89,3 +92,79 @@ func TestGetComponentPorts(t *testing.T) {
 		})
 	}
 }
+
+func TestGetDefaultComponentName(t *testing.T) {
+	tests := []struct {
+		testName           string
+		componentType      string
+		componentPath      string
+		componentPathType  util.ComponentCreateType
+		existingComponents []ComponentInfo
+		wantErr            bool
+		wantRE             string
+		needPrefix         bool
+	}{
+		{
+			testName:           "Case: App prefix not configured",
+			componentType:      "nodejs",
+			componentPathType:  util.GIT,
+			componentPath:      "https://github.com/openshift/nodejs.git",
+			existingComponents: []ComponentInfo{},
+			wantErr:            false,
+			wantRE:             "nodejs-*",
+			needPrefix:         false,
+		},
+		{
+			testName:           "Case: App prefix configured",
+			componentType:      "nodejs",
+			componentPathType:  util.SOURCE,
+			componentPath:      "./testing",
+			existingComponents: []ComponentInfo{},
+			wantErr:            false,
+			wantRE:             "testing-nodejs-*",
+			needPrefix:         true,
+		},
+		{
+			testName:           "Case: App prefix configured",
+			componentType:      "wildfly",
+			componentPathType:  util.BINARY,
+			componentPath:      "./testing.war",
+			existingComponents: []ComponentInfo{},
+			wantErr:            false,
```

Comment:
we should create a test to find an error.

Reply:
Same as in https://github.com/redhat-developer/odo/pull/741#discussion_r226592953